src/urllib3/response.py

```diff
@@ -906,6 +906,10 @@ def url(self) -> Optional[str]:
         else:
             return self._request_url
 
+    @url.setter
+    def url(self, url: str) -> None:
+        self._request_url = url
+
```

Comment:
This is a hacky workaround to allow Sphinx to set the URL:

https://github.com/sphinx-doc/sphinx/blob/564439e66d81693edd08fa5973a14151af976ca8/sphinx/ext/intersphinx.py#L118-L120

"hacky" because it does not work when there are redirects. (We should first agree that whether we want to allow this or not. I think we do.)

Reply:
I think allowing this is fine.